backend/eduschola_project/edusch_app/views.py

```diff
@@ -151,20 +151,15 @@ def updateInstructor(self, request, *args, **kwargs):
         return super().update(request, *args, **kwargs)
 
 
-class CreateSchoolApiView(APIView):
-    def post(self, request):
+class CreateSchoolApiView(generics.CreateAPIView):
+    serializer_class = SchoolSerializer
+    queryset = School.objects.all()
 
+    def post(self, request, *args, **kwargs):
```

Comment:
You don't need to explicitly define the post method in this case, unless you want to add some custom behavior that the CreateAPIView does not provide by default.

Reply:
okay i added it because i wasnt sure CreateAPIView was going to validate the serializer data...i will refactor it now thanks for the review.